src/backend/backend.ts

```diff
@@ -15,39 +14,67 @@ import {
   Message,
   MessageFactory,
   MessageType,
+  browserDispatch,
   browserSubscribe,
 } from '../communication';
 
 import {send} from './indirect-connection';
 
 import {
-  Route,
   MainRoute,
-  defineLookupOperation,
   highlight,
   parseRoutes,
 } from './utils';
 
+import {serialize} from '../utils';
+
+import {MessageQueue} from '../structures';
+
 import {SimpleOptions} from '../options';
 
 declare const ng;
 declare const getAllAngularRootElements: () => Element[];
 declare const treeRenderOptions: SimpleOptions;
 
+/// For tree deltas that contain more changes than {@link deltaThreshold},
+/// we simply send the entire tree again instead of trying to patch it
+/// since it will be faster than trying to apply hundreds or thousands of
+/// changes to an existing tree.
+const deltaThreshold = 512;
```

Comment:
I don't have measurements. But I did do a bit of testing around it and it appeared to improve performance. Logically it also makes sense. There is no sense trying to compare two massive trees that you already know are vastly different. Just send the new one. Other than that does the PR look OK? Did you try running it?